apps/theming-designer/src/components/SemanticSlots.tsx

```diff
@@ -432,33 +44,104 @@ export const SemanticSlots: React.StatelessComponent<ISemanticSlotsProps> = (pro
     );
   };
 
-  const fillVariantSlotsList = (variantType: VariantThemeType, mappingType: {}): JSX.Element[] => {
+  const trimPaletteSlots = (paletteSlots: {}): {} => {
+    let trimmedPaletteSlots = {};
+    for (let palette in paletteSlots) {
+      if (
+        palette.startsWith('theme') ||
+        palette.startsWith('neutral') ||
+        palette === 'black' ||
+        palette === 'white' ||
+        palette === 'redDark' ||
+        palette === 'accent'
+      ) {
+        (trimmedPaletteSlots as any)[palette] = (paletteSlots as any)[palette];
+      }
+    }
+    return trimmedPaletteSlots;
+  };
+
+  const isASemanticColor = (semanticString: string): boolean => {
+    return (
+      semanticString.startsWith('body') ||
+      semanticString.startsWith('disabled') ||
+      semanticString.startsWith('focus') ||
+      semanticString.startsWith('variant') ||
+      semanticString.startsWith('default') ||
+      semanticString.includes('action') ||
+      semanticString.includes('link') ||
+      semanticString.startsWith('accent') ||
+      semanticString.startsWith('list') ||
+      semanticString.startsWith('button') ||
+      semanticString.startsWith('menu') ||
+      semanticString.startsWith('input')
+    );
+  };
+
+  const trimSemanticSlotsOrNames = (semanticSlots: {} | string[]): {} | string[] => {
```

Comment:
this is never a string[] from the usage I can see

Reply:
When this is called in line 140 it gets passed in string[] - Object.keys(semanticSlotsNone)